File: frontend/src/api/pos.ts
import { apiClient } from './client'

export type OrderType = 'counter' | 'delivery'
export type OrderStatus = 'pending' | 'confirmed' | 'preparing' | 'ready' | 'completed' | 'cancelled'
export type PaymentMethod = 'cash' | 'card' | 'transfer'

export interface OrderItem {
  id: number
  product_id: number
  quantity: number
  unit_price: string
  subtotal: string
}

export interface Payment {
  id: number
  order_id: number
  method: PaymentMethod
  amount: string
  reference?: string
  paid_at: string
}

export interface Order {
  id: number
  order_type: OrderType
  status: OrderStatus
  customer_id?: number
  notes?: string
  total_amount: string
  items: OrderItem[]
  payment?: Payment
  created_at: string
  updated_at: string
}

export interface OrderCreate {
  order_type: OrderType
  customer_id?: number
  notes?: string
  items: { product_id: number; quantity: number }[]
  delivery_address?: string
}

export interface PaymentCreate {
  method: PaymentMethod
  amount: number
  reference?: string
}

export const posApi = {
  listOrders: (params?: { skip?: number; limit?: number; status?: OrderStatus }) =>
    apiClient.get<Order[]>('/pos/orders', { params }).then((r) => r.data),

  createOrder: (data: OrderCreate) =>
    apiClient.post<Order>('/pos/orders', data).then((r) => r.data),

  getOrder: (id: number) =>
    apiClient.get<Order>(`/pos/orders/${id}`).then((r) => r.data),

  updateStatus: (id: number, status: OrderStatus) =>
    apiClient.patch<Order>(`/pos/orders/${id}/status`, { status }).then((r) => r.data),

  recordPayment: (orderId: number, data: PaymentCreate) =>
    apiClient.post<Payment>(`/pos/orders/${orderId}/payment`, data).then((r) => r.data),
}

File: backend/app/schemas/pos.py
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel, ConfigDict
from app.models.order import OrderType, OrderStatus, PaymentMethod


class OrderItemCreate(BaseModel):
    product_id: int
    quantity: int


class OrderItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    product_id: int
    quantity: int
    unit_price: Decimal
    subtotal: Decimal


class OrderCreate(BaseModel):
    order_type: OrderType
    customer_id: int | None = None
    notes: str | None = None
    items: list[OrderItemCreate]
    # Required when order_type == delivery
    delivery_address: str | None = None


class OrderStatusUpdate(BaseModel):
    status: OrderStatus


class PaymentCreate(BaseModel):
    method: PaymentMethod
    amount: Decimal
    reference: str | None = None


class PaymentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    order_id: int
    method: PaymentMethod
    amount: Decimal
    reference: str | None
    paid_at: datetime


class OrderResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    order_type: OrderType
    status: OrderStatus
    customer_id: int | None
    notes: str | None
    total_amount: Decimal
    items: list[OrderItemResponse] = []
    payment: PaymentResponse | None = None
    created_at: datetime
    updated_at: datetime
